lib/menu.c

```diff
@@ -828,6 +828,25 @@ bm_menu_run_with_key(struct bm_menu *menu, enum bm_key key, uint32_t unicode)
             }
             break;
 
+        case BM_KEY_PASTE:
+            {
+                FILE *clipboard;
+                clipboard = popen("wl-paste -t text/plain", "r");
+                if (clipboard == NULL) {
+                    clipboard = popen("xclip -t text/plain -out", "r");
+                }
+                if (clipboard == NULL) {
+                    break;
+                }
+                int c;
+                while ((c = fgetc(clipboard)) != EOF) {
```

Comment:
probably could be done faster by reading to a buffer, but not worth it unless this becomes a real issue.